projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java

```diff
@@ -226,17 +227,21 @@ public static int executeMain(File file, Object... args) {
         LLVMLogger.info("current file: " + file.getAbsolutePath());
         Source fileSource;
         try {
-            fileSource = Source.fromFileName(file.getAbsolutePath());
+            fileSource = Source.newBuilder(file).build();
             return evaluateFromSource(fileSource, args);
         } catch (IOException e) {
             throw new AssertionError(e);
         }
     }
 
     public static int executeMain(String codeString, Object... args) {
-        Source fromText = Source.fromText(codeString, "code string").withMimeType(LLVMLanguage.LLVM_IR_MIME_TYPE);
-        LLVMLogger.info("current code string: " + codeString);
-        return evaluateFromSource(fromText, args);
+        try {
+            Source fromText = Source.newBuilder(codeString).mimeType(LLVMLanguage.LLVM_IR_MIME_TYPE).build();
+            LLVMLogger.info("current code string: " + codeString);
+            return evaluateFromSource(fromText, args);
+        } catch (MissingNameException e) {
+            throw new AssertionError(e);
+        }
```

Comment:
Thanks! I did not realize that this method was not used (or just by a single test case that is not part of the CI). I will remove the method altogether.